### PR TITLE
Extend submenu background variables to 90100

### PR DESCRIPTION
**Pull request by Ignoble61**
I know you're busy (and I hope everything has gone well :smile:), so this is fulfilling a forum request for the submenu backgrounds to work for vertical submenu.

No-one expects you to test or merge this quickly :wink:

### CODE REVIEW

**Comment by Ignoble61** [comment]
Added a second commit which removes script.grab.fanart backgrounds and replaces them (along with additional background options) with skin helper service backgrounds (requires re-settings random movies/episodes/music backgrounds). Still waiting to hear that Mini Silvo has arrived safely :grin:

**Comment by mikesilvo164** [comment]
Hi, Bob!
Everything went well with the Mrs. and baby Jessica was born yesterday at 1:45pm.

My only issues is with removing script.grab.fanart as I do not want skin.helper.service as a requirement. I want to remove a lot of the existing requirements and have them be installed when needed. The colored studios for example can be removed and added on demand using RunPlugin for Jarvis and Isengard and the new InstallAddon for K**** as a conditional on click when activating the colored flags setting. The properties for the images returned by skin.helper seem to return paths instead of links to cached items and that might be a performance hit in certain situations like a user having a nas that would need to spin up.

Do you know how those properties are filled and when called if they are using the images that should already be in the cache?

**Comment by Ignoble61** [comment]
Congratulations to you and Mrs Silvo on the birth of Mini Silvo :)

Now, go enjoy some rest whilst you can. Seriously, what are you doing even looking at things like this? If this is your first, you'll come to regret it. If it's not your first, you should know better :wink: 

I'm not a huge fan of skin helper myself, as one of the two devices we use Kodi on in my household is a rPi1 - not a good device to be running that script (or any unnecessary script) on. Which is what lead me to changes - which would need implementing by @mistervee for Nox 5 - to how the skin knows what widget providing scripts are installed, and lets additional scripts be installed.

Documentation here:- https://github.com/Ignoble61/script.skinshortcuts/blob/master/resources/docs/advanced/Managing%20widgets.md#default-widgets-from-skin-helper-service.

Basically, with this method - and assuming none of the additional skin helper methods are necessary for the script - that script, plus Extended Info, plus Library Data Provider are completely optional and are all installable via Skin Shortcuts whilst selecting a widget if the user so desires (and, if other skins play along, are deletable just as easily.)

First, take a break - enjoy any sleep you can get before the sleepless nights, and so on. Then, ask for any more details if you need them :)

A huge congratulations again. And, because apparently I forget this a week ago, a merry festivus to you :wink:

**Comment by mikesilvo164** [comment]
Thanks Bob :grinning: It is my second and I am typing this to you using my phone while Jessica is sleeping.  I can survive on a couple of hours of sleep so I'm all good and this one will be a lot easier than the first cause I kinda know what I am doing now :wink:

I think random library backgrounds are a fundamental feature and skin.helper providing so many things makes it a script that I think belongs in the optional category and not used for providing (imho required) library fanart for the home menu. It is a great script but I am hesitant to use it for scripts that have been specifically made for an important feature. I assume the overhead of skin.helper would be more demanding than grab.fanart, also not sure how optimized the properties are since I see paths returned and I do not even use that kind of thing to display extra fanart.

Trying my best to avoid user issues since support is one of the things I already feel like I spend too much time on :frowning:

**Comment by Ignoble61** [comment]
If its your second you know you really should be getting some sleep. But as I never did with any of mine, I'm really not going to push the issue ;)

Yes, I agree Skin Helper should be optional - and, with reference to the link I gave in my previous comment, with Skin Shortcuts default widgets/backgrounds it is optional - with links given to the user (either the 'Get More' button in a select dialog or - and I forget the exact wording - an 'Enable default widgets' option - when using the default widget-groupings, or an override which enables the relevant options.)

Having said that, whilst I appreciate the work that has gone into the skin helper script, I'm with you that it probably does too much (especially, too much that is actually processor intensive) for a single script whereby its features (and therefore its cpu usage is all or nothing), so say the word and I'll remove the skin-helper backgrounds or re-instate the library.fanart backgrounds, with the skin.helper being optional.

Your skin, your choice ;)

**Comment by mikesilvo164** [comment]
Thanks! :smile:
